### PR TITLE
Add description metadata

### PR DESCRIPTION
**Pull request by kmclarnon**
This seemed like a decent way forward here.  It doesn't touch much of the code so I don't see it being difficult to maintain in the future.

@stevegutz @Xcelled

### CODE REVIEW

File: check_api/src/main/java/com/google/errorprone/matchers/Description.java [hunk 1]
```diff
@@ -29,7 +29,11 @@
 import com.sun.source.tree.Tree;
 import com.sun.tools.javac.tree.JCTree;
 import com.sun.tools.javac.util.JCDiagnostic.DiagnosticPosition;
+
```
Comment (stevegutz) [inline check_api/src/main/java/com/google/errorprone/matchers/Description.java:32]: nit: accidental newline? Or do we just let them come in?

File: check_api/src/main/java/com/google/errorprone/matchers/Description.java [hunk 2]
```diff
@@ -164,6 +180,7 @@ private Builder(
       this.linkUrl = linkUrl;
       this.severity = Preconditions.checkNotNull(severity);
       this.rawMessage = Preconditions.checkNotNull(rawMessage);
+      this.metadata = new HashMap<>();
```
Comment (stevegutz) [inline check_api/src/main/java/com/google/errorprone/matchers/Description.java:183]: nit: `LinkedHashMap`? I love maintaining the order 😄

**Review by stevegutz — COMMENTED** [review]
I wonder also about building up an ImmutableMap 🤷‍♂️

File: check_api/src/main/java/com/google/errorprone/matchers/Description.java (hunk 2 above)
Comment (kmclarnon) [reply check_api/src/main/java/com/google/errorprone/matchers/Description.java:183]: Haha sure I can change that

File: check_api/src/main/java/com/google/errorprone/matchers/Description.java (hunk 1 above)
Comment (kmclarnon) [reply check_api/src/main/java/com/google/errorprone/matchers/Description.java:32]: hmm, not sure where that came from, I can remove

**Comment by kmclarnon** [comment]
I originally chose not to use an immutable map because I didn't see guava listed as a dep, but it seems it is so I can change that

**Pull request merged**